src/System.Net.HttpListener/tests/SimpleHttpTests.cs

```diff
@@ -32,19 +30,154 @@ public static void Supported_True()
+        public void BasicTest_AbortNoStart_NoException()
+        {
+            HttpListener listener = new HttpListener();
+            listener.Abort();
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public void BasicTest_StartThrowsAbortCalledInFinally_AbortDoesntThrow()
+        {
+            HttpListener listener = new HttpListener();
+            // try to add an invalid prefix (invalid port number)
+            listener.Prefixes.Add("http://doesntexist:99999999/");
+            try
+            {
+                Assert.Throws<HttpListenerException>(() => listener.Start());
+            }
+            finally
+            {
+                // even though listener wasn't started (state is 'Closed'), Abort() must not throw.
+                listener.Abort();
+
+                // calling Abort() again should still not throw.
+                listener.Abort();
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public async Task RequestTransferEncoding_MixedCaseChunked_Success()
+        {
+            Task<HttpListenerContext> serverContext = _listener.GetContextAsync();
+
+            using (HttpClient client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.TransferEncodingChunked = true;
+                Task<HttpResponseMessage> clientTask = client.PostAsync(_factory.ListeningUrl, new StringContent("Z"));
+
+                HttpListenerContext context = await serverContext;
+                Assert.Equal(-1, context.Request.ContentLength64);
+                Assert.Equal("chunked", context.Request.Headers["Transfer-Encoding"]);
+                byte[] buffer = new byte[10];
+                Assert.Equal(1, context.Request.InputStream.Read(buffer, 0, buffer.Length));
+                Assert.Equal((byte)'Z', buffer[0]);
+                context.Response.Close();
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public async Task UnknownHeaders_Success()
+        {
+            Task<HttpListenerContext> server = _listener.GetContextAsync();
+
+            using (HttpClient client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.ConnectionClose = true;
+
+                HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Get, _factory.ListeningUrl);
+
+                for (int i = 0; i < 1000; i++)
+                {
+                    requestMessage.Headers.Add($"custom{i}", i.ToString());
+                }
+
+                Task<HttpResponseMessage> responseMessage = client.SendAsync(requestMessage);
+                HttpListenerContext context = await server;
+
+                Assert.Contains("999", context.Request.Headers["custom999"]);
```

Comment:
Shouldn't we validate that all of the added headers were received, not just one of them?